vulcan/lib/server.rb

```diff
@@ -16,7 +16,7 @@ class Server < Etna::Server
     end
 
     get 'api/:project_name/workflows', action: 'workflows#fetch', as: :workflows_view, auth: { user: { can_view?: :project_name } }
```

Comment:
Does there need to be an additional route to get the steps for a given workflow? Or is that considered "data" somehow, and the UI will be passed the cell_hash for retrieving that?

Reply:
So, just to be super clear, I imagine that you will use, abstractly, these apis:

1.  workflow route to get, the steps / workflow (minus any states, just definitions of them).
2.  status / input routes to add new user inputs & get the current state of work (which is synchronous in this version, so it basically will run everything and give you the status of those completed jobs)
3.  data endpoints, referenced in the status values, of where to stream each step's outputs.

Reply:
Okay, cool that makes sense. At some point once we move beyond just UMAP, I think we'll also need a "here are the workflows you can access" API?

Reply:
Absolutely.  Basically, there should be a crud interface to listing, creating, editing, and fetching workflows.  @graft also wants it to be possible to adhoc edit workflows, as well, which is doable with a backing crud.  But that is beyond umap.